src/class/HTTPRequest/HTTPRequest.hpp

```diff
@@ -0,0 +1,78 @@
+
+#ifndef H_T_T_P_REQUEST_HPP
+#define H_T_T_P_REQUEST_HPP
+
+#include <BasicHTTPRequest/BasicHTTPRequest.hpp>
+#include <HTTPRequestHandler/HTTPRequestHandler.hpp>
+#include <Server/Server.hpp>
+#include <iostream>
+#include <map>
+#include <string>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#ifdef TEST_ON
+#define private public
+#define protected public
+#endif
+
+class HTTPRequest
```

Comment:
since this class is about the meta data of an http req and is about reading and writeing a rsponce i think the name should be sometinhg like `HTTPCall`